src/logic/safe/store/actions/createTransaction.ts

```diff
@@ -239,6 +243,8 @@ export class TxSender {
     // On-chain signature or execution
     try {
       await this.sendTx(confirmCallback)
+      // Only track if transaction succeeds
+      trackEvent(WALLET_EVENTS.ON_CHAIN_INTERACTION)
```

Comment:
Can't both of the events be called from this.onComplete depending on the signature?

Reply:
Very true